cpp/src/arrow/type.h

```diff
@@ -1517,6 +1517,12 @@ class ARROW_EXPORT FieldRef {
   std::string ToString() const;
 
   size_t hash() const;
+  struct Hash {
+    size_t operator()(const FieldRef& ref) const { return ref.hash(); }
+  };
+
+  explicit operator bool() const { return Equals(FieldPath{}); }
```

Comment:
I'm not sure I understand the semantics just by reading the code. Could you add a comment above?

Reply:
(or rename this `bool empty() const`?)

Reply:
I'll rename to `empty()`